File: sprint3/By languages/C#/lesson7code1.cs
using System;
using System.Collections.Generic;

class Program {
    static void Main(string[] args) {
        var array = new List<int> { 5, 2, 4, 6, 1, 3 };
        InsertionSort(array);
    }

    static void InsertionSort(List<int> array) {
        for (int i = 1; i < array.Count; i++) {
            int item_to_insert = array[i];
            int j = i;
            while (j > 0 && item_to_insert < array[j - 1]) {
                array[j] = array[j - 1];
                j--;
            }
            array[j] = item_to_insert;
            Console.WriteLine($"step {i}, sorted {i + 1} elements: {string.Join(" ", array.GetRange(0, i + 1))}");
        }
    }
}

File: sprint3/By languages/swift/lesson7code1.swift
func insertionSort(_ array: inout [Int]) {
    for i in 1..<array.count {
        let itemToInsert = array[i]
        var j = i
        while j > 0 && itemToInsert < array[j - 1] {
            array[j] = array[j - 1]
            j -= 1
        }
        array[j] = itemToInsert
        print("step \(i), sorted \(i + 1) elements: \(array)")
    }
}

var array = [11, 2, 9, 7, 1]
insertionSort(&array)
// step 1, sorted 2 elements: [2, 11, 9, 7, 1]
// step 2, sorted 3 elements: [2, 9, 11, 7, 1]
// step 3, sorted 4 elements: [2, 7, 9, 11, 1]
// step 4, sorted 5 elements: [1, 2, 7, 9, 11]